service/history/execution/mutable_state_builder.go

```diff
@@ -485,64 +485,31 @@ func (e *mutableStateBuilder) FlushBufferedEvents() error {
 		}
 	}
 
-	// Sometimes we see buffered events are out of order when read back from database.  This is mostly not an issue
-	// except in the Activity case where ActivityStarted and ActivityCompleted gets out of order.  The following code
-	// is added to reorder buffered events to guarantee all activity completion events will always be processed at the end.
-	var reorderedEvents []*types.HistoryEvent
-	reorderFunc := func(bufferedEvents []*types.HistoryEvent) {
-		for _, event := range bufferedEvents {
-			switch event.GetEventType() {
-			case types.EventTypeActivityTaskCompleted,
-				types.EventTypeActivityTaskFailed,
-				types.EventTypeActivityTaskCanceled,
-				types.EventTypeActivityTaskTimedOut:
-				reorderedEvents = append(reorderedEvents, event)
-			case types.EventTypeChildWorkflowExecutionCompleted,
-				types.EventTypeChildWorkflowExecutionFailed,
-				types.EventTypeChildWorkflowExecutionCanceled,
-				types.EventTypeChildWorkflowExecutionTimedOut,
-				types.EventTypeChildWorkflowExecutionTerminated:
-				reorderedEvents = append(reorderedEvents, event)
-			default:
-				newCommittedEvents = append(newCommittedEvents, event)
-			}
-		}
-	}
-
-	e.logDuplicatedActivityEvents(newBufferedEvents, "newBufferedEvents")
-
 	// no decision in-flight, flush all buffered events to committed bucket
 	if !e.HasInFlightDecision() {
-		// adding logs to help identify duplicate activity task events
-		// duplicated activity events can cause DecisionTaskFailed events with cause UNHANDLED_DECISION
-		// and cause workflow to be stuck in decision task failed state
-		// this can be removed after the root cause is identified and fixed
-		// TODO: remove this after the root cause is identified and fixed or add deduplication
-		e.logDuplicatedActivityEvents(e.bufferedEvents, "bufferedEvents")
-		e.logDuplicatedActivityEvents(e.updateBufferedEvents, "updateBufferedEvents")
+		var allBufferedEvents []*types.HistoryEvent
 
 		// flush persisted buffered events
 		if len(e.bufferedEvents) > 0 {
-			reorderFunc(e.bufferedEvents)
+			allBufferedEvents = append(allBufferedEvents, e.bufferedEvents...)
 			e.bufferedEvents = nil
 		}
 		if e.hasBufferedEventsInDB {
 			e.clearBufferedEvents = true
 		}
-
 		// flush pending buffered events
-		reorderFunc(e.updateBufferedEvents)
-		// clear pending buffered events
+		allBufferedEvents = append(allBufferedEvents, e.updateBufferedEvents...)
```

Comment:
nit, not super sure why this isn't wrapped in a check like 493, should they be the same?
```
if len(e.updateBufferedEvents) >0 {
    allBufferedEvents = append(allBufferedEvents, e.updateBufferedEvents...)
} 
```